src/main/java/world/bentobox/bentobox/database/AbstractDatabaseHandler.java

```diff
@@ -100,26 +101,20 @@ protected AbstractDatabaseHandler(BentoBox plugin, Class<T> type, DatabaseConnec
         if (!plugin.isEnabled()) return;
         // Run async queue
         processQueue = new ConcurrentLinkedQueue<>();
-        asyncSaveTask = Bukkit.getScheduler().runTaskAsynchronously(plugin, () -> {
-            // Loop continuously
-            while (!shutdown || !processQueue.isEmpty()) {
-                while (!processQueue.isEmpty()) {
+        asyncSaveTask = Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, () -> {
+            // Check shutdown
+            if(shutdown || plugin.isShutdown()) {
+                // Cancel - this will only get called if the plugin is shutdown separately to the server
+                databaseConnector.closeConnection(dataObject);
+                asyncSaveTask.cancel();
+            } else if (!inSave && !processQueue.isEmpty()) {
+                inSave = true;
+                while(!processQueue.isEmpty()) {
                     processQueue.poll().run();
                 }
-                // Shutdown flag
-                shutdown = plugin.isShutdown();
-                // Clear the queue and then sleep
-                try {
-                    Thread.sleep(25);
-                } catch (InterruptedException e) {
-                    plugin.logError("Thread sleep error " + e.getMessage());
-                    Thread.currentThread().interrupt();
-                }
+                inSave = false;
```

Comment:
The queue processor now uses the Bukkit async timer repeating task instead of a raw async task with a sleep. This task will automatically get canceled when the server is shutting down. It also means I do not have to use Thread.sleep(). The queue is cleared every tick, if possible.